image-hd.c

```diff
@@ -610,7 +610,7 @@ static int hdimage_setup(struct image *image, cfg_t *cfg)
 				return -EINVAL;
 			}
 		}
-		if (hd->gpt) {
+		if (hd->gpt && part->in_partition_table) {
```

Comment:
Right, I've been working on cleaning this up (see  #138) but that's not finished either. We'll see which PR is done first. Then the other needs to be rebased.